### ISSUE TITLE
Segmentation Fault (Core dumped) on yolov3-tiny_xnor

### ISSUE THREAD

**Issue by Hugh-Chang** [AlexeyAB/darknet#6869]
Hello @AlexeyAB Alexey, 
I'm facing this problem. 
It is fine when I trained yolov3-tiny model with coco data. 

./darknet detector train cfg/coco.data cfg/yolov3-tiny.cfg yolov3-tiny.conv.15
 GPU isn't used 
 OpenCV isn't used - data augmentation will be slow 
yolov3-tiny
mini_batch = 4, batch = 64, time_steps = 1, train = 1 
   layer   filters  size/strd(dil)      input                output
   0 conv     16       3 x 3/ 1    416 x 416 x   3 ->  416 x 416 x  16 0.150 BF
   1 max                2x 2/ 2    416 x 416 x  16 ->  208 x 208 x  16 0.003 BF
   2 conv     32       3 x 3/ 1    208 x 208 x  16 ->  208 x 208 x  32 0.399 BF
   3 max                2x 2/ 2    208 x 208 x  32 ->  104 x 104 x  32 0.001 BF
   4 conv     64       3 x 3/ 1    104 x 104 x  32 ->  104 x 104 x  64 0.399 BF
   5 max                2x 2/ 2    104 x 104 x  64 ->   52 x  52 x  64 0.001 BF
   6 conv    128       3 x 3/ 1     52 x  52 x  64 ->   52 x  52 x 128 0.399 BF
   7 max                2x 2/ 2     52 x  52 x 128 ->   26 x  26 x 128 0.000 BF
   8 conv    256       3 x 3/ 1     26 x  26 x 128 ->   26 x  26 x 256 0.399 BF
   9 max                2x 2/ 2     26 x  26 x 256 ->   13 x  13 x 256 0.000 BF
  10 conv    512       3 x 3/ 1     13 x  13 x 256 ->   13 x  13 x 512 0.399 BF
  11 max                2x 2/ 1     13 x  13 x 512 ->   13 x  13 x 512 0.000 BF
  12 conv   1024       3 x 3/ 1     13 x  13 x 512 ->   13 x  13 x1024 1.595 BF
  13 conv    256       1 x 1/ 1     13 x  13 x1024 ->   13 x  13 x 256 0.089 BF
  14 conv    512       3 x 3/ 1     13 x  13 x 256 ->   13 x  13 x 512 0.399 BF
  15 conv    255       1 x 1/ 1     13 x  13 x 512 ->   13 x  13 x 255 0.044 BF
  16 yolo
[yolo] params: iou loss: mse (2), iou_norm: 0.75, obj_norm: 1.00, cls_norm: 1.00, delta_norm: 1.00, scale_x_y: 1.00
  17 route  13 		                           ->   13 x  13 x 256 
  18 conv    128       1 x 1/ 1     13 x  13 x 256 ->   13 x  13 x 128 0.011 BF
  19 upsample                 2x    13 x  13 x 128 ->   26 x  26 x 128
  20 route  19 8 	                           ->   26 x  26 x 384 
  21 conv    256       3 x 3/ 1     26 x  26 x 384 ->   26 x  26 x 256 1.196 BF
  22 conv    255       1 x 1/ 1     26 x  26 x 256 ->   26 x  26 x 255 0.088 BF
  23 yolo
[yolo] params: iou loss: mse (2), iou_norm: 0.75, obj_norm: 1.00, cls_norm: 1.00, delta_norm: 1.00, scale_x_y: 1.00
Total BFLOPS 5.571 
avg_outputs = 341534 
Loading weights from yolov3-tiny.conv.15...
 seen 64, trained: 0 K-images (0 Kilo-batches_64) 
Done! Loaded 15 layers from weights-file 
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
 Detection layer: 16 - type = 28 
 Detection layer: 23 - type = 28 
Resizing, random_coef = 1.40 

 608 x 608 
 Create 64 permanent cpu-threads 
Loaded: 1.962383 seconds
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 16 Avg (IOU: 0.289123, GIOU: 0.147824), Class: 0.552319, Obj: 0.463194, No Obj: 0.533012, .5R: 0.100000, .75R: 0.000000, count: 10, class_loss = 383.822876, iou_loss = 15.701660, total_loss = 399.524536 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 23 Avg (IOU: 0.272068, GIOU: -0.004992), Class: 0.509732, Obj: 0.464775, No Obj: 0.514948, .5R: 0.125000, .75R: 0.000000, count: 8, class_loss = 1270.804810, iou_loss = 12.006104, total_loss = 1282.810913 
 total_bbox = 18, rewritten_bbox = 0.000000 % 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 16 Avg (IOU: 0.235575, GIOU: 0.149125), Class: 0.479495, Obj: 0.432390, No Obj: 0.534352, .5R: 0.000000, .75R: 0.000000, count: 12, class_loss = 399.649597, iou_loss = 12.600067, total_loss = 412.249664 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 23 Avg (IOU: 0.133382, GIOU: -0.335024), Class: 0.742346, Obj: 0.526544, No Obj: 0.515321, .5R: 0.000000, .75R: 0.000000, count: 1, class_loss = 1234.550659, iou_loss = 2.316406, total_loss = 1236.867065 
 total_bbox = 31, rewritten_bbox = 0.000000 % 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 16 Avg (IOU: 0.334253, GIOU: 0.219245), Class: 0.472885, Obj: 0.526311, No Obj: 0.535243, .5R: 0.076923, .75R: 0.000000, count: 13, class_loss = 401.109711, iou_loss = 9.158905, total_loss = 410.268616 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 23 Avg (IOU: 0.328976, GIOU: 0.142247), Class: 0.561654, Obj: 0.457793, No Obj: 0.515030, .5R: 0.125000, .75R: 0.000000, count: 8, class_loss = 1268.266357, iou_loss = 6.396118, total_loss = 1274.662476 
 total_bbox = 52, rewritten_bbox = 0.000000 % 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 16 Avg (IOU: 0.381079, GIOU: 0.210374), Class: 0.422930, Obj: 0.598679, No Obj: 0.534744, .5R: 0.214286, .75R: 0.000000, count: 14, class_loss = 407.216095, iou_loss = 8.494690, total_loss = 415.710785 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 23 Avg (IOU: 0.373418, GIOU: 0.327793), Class: 0.428257, Obj: 0.533370, No Obj: 0.514343, .5R: 0.111111, .75R: 0.000000, count: 9, class_loss = 1273.297119, iou_loss = 6.855225, total_loss = 1280.152344 
 total_bbox = 75, rewritten_bbox = 0.000000 % 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 16 Avg (IOU: 0.270663, GIOU: 0.185042), Class: 0.566725, Obj: 0.504355, No Obj: 0.535474, .5R: 0.181818, .75R: 0.000000, count: 11, class_loss = 395.707611, iou_loss = 12.820404, total_loss = 408.528015 
v3 (mse loss, Normalizer: (iou: 0.75, cls: 1.00) Region 23 Avg (IOU: 0.238283, GIOU: -0.015401), Class: 0.445951, Obj: 0.527603, No Obj: 0.515958, .5R: 0.100000, .75R: 0.000000, count: 20, class_loss = 1333.578369, iou_loss = 20.936401, total_loss = 1354.514771 

However, I used yolov3-tiny_xnor.cfg with coco to train model, it occurs Segmentation Fault (core dumped). I tried to reduce the batch and subdivision size but it still happens. The problem is shown as below. 

./darknet detector train cfg/coco.data cfg/yolov3-tiny_xnor.cfg yolov3-tiny.conv.15
 GPU isn't used 
 OpenCV isn't used - data augmentation will be slow 
yolov3-tiny_xnor
mini_batch = 2, batch = 64, time_steps = 1, train = 1 
   layer   filters  size/strd(dil)      input                output
   0 conv     16       3 x 3/ 1    416 x 416 x   3 ->  416 x 416 x  16 0.150 BF
   1 max                2x 2/ 2    416 x 416 x  16 ->  208 x 208 x  16 0.003 BF
   2 convXB   32       3 x 3/ 1    208 x 208 x  16 ->  208 x 208 x  32 0.012 BF
   3 max                2x 2/ 2    208 x 208 x  32 ->  104 x 104 x  32 0.001 BF
   4 convXB   64       3 x 3/ 1    104 x 104 x  32 ->  104 x 104 x  64 0.012 BF
   5 max                2x 2/ 2    104 x 104 x  64 ->   52 x  52 x  64 0.001 BF
   6 convXB  128       3 x 3/ 1     52 x  52 x  64 ->   52 x  52 x 128 0.012 BF
   7 max                2x 2/ 2     52 x  52 x 128 ->   26 x  26 x 128 0.000 BF
   8 convX   256       3 x 3/ 1     26 x  26 x 128 ->   26 x  26 x 256 0.012 BF
   9 max                2x 2/ 2     26 x  26 x 256 ->   13 x  13 x 256 0.000 BF
  10 convXB  512       3 x 3/ 1     13 x  13 x 256 ->   13 x  13 x 512 0.012 BF
  11 max                2x 2/ 1     13 x  13 x 512 ->   13 x  13 x 512 0.000 BF
  12 convXB 1024       3 x 3/ 1     13 x  13 x 512 ->   13 x  13 x1024 0.050 BF
  13 convX   256       1 x 1/ 1     13 x  13 x1024 ->   13 x  13 x 256 0.003 BF
  14 conv    512       3 x 3/ 1     13 x  13 x 256 ->   13 x  13 x 512 0.399 BF
  15 conv    255       1 x 1/ 1     13 x  13 x 512 ->   13 x  13 x 255 0.044 BF
  16 yolo
[yolo] params: iou loss: mse (2), iou_norm: 0.75, obj_norm: 1.00, cls_norm: 1.00, delta_norm: 1.00, scale_x_y: 1.00
  17 route  13 		                           ->   13 x  13 x 256 
  18 convX   128       1 x 1/ 1     13 x  13 x 256 ->   13 x  13 x 128 0.000 BF
  19 upsample                 2x    13 x  13 x 128 ->   26 x  26 x 128
  20 route  19 8 	                           ->   26 x  26 x 384 
  21 convX   256       3 x 3/ 1     26 x  26 x 384 ->   26 x  26 x 256 0.037 BF
  22 conv    255       1 x 1/ 1     26 x  26 x 256 ->   26 x  26 x 255 0.088 BF
  23 yolo
[yolo] params: iou loss: mse (2), iou_norm: 0.75, obj_norm: 1.00, cls_norm: 1.00, delta_norm: 1.00, scale_x_y: 1.00
Total BFLOPS 0.839 
avg_outputs = 341534 
Loading weights from yolov3-tiny.conv.15...
 seen 64, trained: 0 K-images (0 Kilo-batches_64) 
Done! Loaded 15 layers from weights-file 
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
 Detection layer: 16 - type = 28 
 Detection layer: 23 - type = 28 
 Create 64 permanent cpu-threads 
Loaded: 1.683310 seconds
Segmentation fault (core dumped)

Thank you so much for your help.
